src/utils/chartJs/generateCustomTooltip.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import CustomTooltip from './CustomTooltip';
+
+const generateCustomTooltip = (canvas, tooltipModel, series) => {
```

Comment:
Use JSX to construct this `tooltip-container` div.  You should be able to set all the style properties to ensure it is properly positioned.


    <div id="tooltip-container' style={{top: positionY + tooltipModel.caretY - heightTooltip, ...}}>
        <CustomTooltip tooltipModel={tooltipModel} series={series} />
    </div>

the `style` property accepts a object; notice the double curly braces (one for code, the other for object creation)

then, you should be able to inline `CustomTooltip` to result in a single component

Reply:
For this suggestion, please check comment at the bottom of the main thread.